### PR TITLE
dump file, validation on transaction set, group, interchange

### PR DESCRIPTION
**Pull request by mfdeveloper508**
- printing file
- validation on transaction set, group, interchange
- adding file object
- special reader for large file (reading by each interchange)

### CODE REVIEW

File: pkg/file/file.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	for _, change := range f.Interchanges {
+		if err := change.Validate(f.rule); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f File) Print(w io.Writer) {
```
Comment (adamdecaf) [inline pkg/file/file.go:64]: Can we comment about how Print returns a human readable version?

File: pkg/file/interchange.go [hunk 2]
```diff
@@ -154,3 +180,70 @@ func (r Interchange) String(args ...string) string {
 
 	return buf.String()
 }
+
+func (r Interchange) Print(w io.Writer) {
```
Comment (adamdecaf) [inline pkg/file/interchange.go:199]: Same comment here about human-readable output.

File: pkg/file/reader.go [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	"strings"
+
+	"github.com/moov-io/x12/pkg/util"
+)
+
+type Scanner struct {
+	scan *bufio.Scanner
+}
+
+func (b *Scanner) GetInterChange() string {
```
Comment (adamdecaf) [inline pkg/file/reader.go]: Does ASC X12 call it "InterChange"? Usually it would be written `GetInterchange()` as "interchange" is one word.

**Comment by mfdeveloper508** [comment]
Thanks @adamdecaf 
Added cli feature too

**Comment by adamdecaf** [comment]
Awesome, thanks!

**Pull request merged**